### PR TITLE
Controller support

### PR DESCRIPTION
**Pull request by Bram-Boris**
# Description

This PR adds controller support for players.
When using a controller the keyboard controls are disabled for that player.
Hotplug is supported, the controller will be assigned to the first player without one.

Use the controller branch for testing

Bindings:
Left control stick for moving left and right

| XBOX        | PS4           | NINTENDO  | action |
| ------------- | ------------- | ----- | ------------- |
| A  | Cross | B | Jump |
| X  | Square      |   Y | Shoot |
| B | Circle      |    A | Grab |

# How can this be tested?
Connect a controller and try to play! I have tested 3 which should be sufficient

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MarkMeer — COMMENTED** [review]
For me there is a little bit of lag when walking with the keyboard. It feels a bit like you have to wait very short before the key stroke gets detected. Could also just be my Laptop as you said during the project lesson that isn't as smooth in running the game. I'll leave that up to @Peter-JanGootzen . Everything else is good 👍 .

File: lib/systems/movement_system.cpp [hunk 1]
```diff
@@ -28,25 +28,20 @@ void MovementSystem::update(double) {
...
-                vx += TERMINAL_VELOCITY * MOVEMENT_FORCE / mass;
-                if (vx > TERMINAL_VELOCITY / mass) {
-                    vx = TERMINAL_VELOCITY / mass;
-                }
-            } else {
+            vx = x_movement * MOVEMENT_FORCE / mass;
+        }
+        else if (x_movement > 0) {
+            if (vx < 0) 
                 vx = 0;
```
Comment (Peter-JanGootzen) [inline lib/systems/movement_system.cpp]: You can put it one line like on line 33

**Review by Peter-JanGootzen — COMMENTED** [review]
The current controller mappings don't really feel good for a game like this imo.
Only using the left part of the buttons is quite weird. Since we only have 3 actions bound to buttons, I would suggest using the bottom part of the buttons. Have Jump on B(Nintendo), Grab on Y(Nintendo) and Shoot on A(Nintendo). Imo this plays way more relaxed and suits our **_FAST PACED GAMEPLAY_** better.

**Comment by markJavab** [comment]
Well done 👍

**Review by MarkMeer — APPROVED** [review]
Well done 👍

**Review by Peter-JanGootzen — APPROVED** [review]
:shrimp:

**Pull request merged**